### PR TITLE
Add Introduction Page

### PR DESCRIPTION
**Pull request by CoWinkKeyDinkInc**
Signed-off-by: Patrick <cowinkkeydinkinc@gmail.com>

Adds an Introduction Page to the Modules section. This introduction gives a short history and explanation of what an XML file is and what it can do. It links to the Airship Battles example which already explains nicely how a simple map is structured.

There's still a few more changes I want to do, expand on the history and make the introduction the default page when opening the modules tab, which I couldn't figure out on this small laptop.

### CODE REVIEW

File: docs/modules/general/introduction.mdx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+---
+id: introduction
+title: Introduction
+---
+
+### History
+
+PGM originally started off in 2012 as a single game plugin specificaly built to run one map, Airship Battle.
```
Comment (Pugzy) [inline docs/modules/general/introduction.mdx:8]: specifically

File: docs/modules/general/introduction.mdx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+This plugin could manage the match and reset the map easily at the end of the game.
+Other maps were created later, but these all had to be hardcoded into the plugin.
+This was cumbersome and complicated, and there needed to be a new way to quickly program maps into the plugin using an easy to understand language and syntax.
+This problem would later meet its solution, the PvP Game Manager.
+Players unknown to Java programming could now create their own games and matches using an XML file that PGM takes as an input.
+As of now, there are around 1000+ maps made for PGM!
```
Comment (Pugzy) [inline docs/modules/general/introduction.mdx:14]: `around 1000+` doesn't seem quite right, maybe `over 1000`

File: docs/modules/general/introduction.mdx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+### How it Works
+
+All maps made for PGM are programmed using an XML file called `map.xml`.
+XML is structured and layered silimar to HTML, making it easily readable.
```
Comment (Pugzy) [inline docs/modules/general/introduction.mdx:19]: similar

File: docs/modules/general/introduction.mdx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+Everything from objectives, spawn kits, filters, jump pads, build areas, kill rewards and damage disable is defined in a map's XML file.
+Most of the XML can be programmed with little involvement of Minecraft, save for testing and getting coordinates.
+PGM is also backwards compatible with XML files that were made for a previous spec (map proto) so older maps don't need to be upgraded to be playable.
+
+Aspects to a map are defined by modules, and information can be placed inside and in-between modules.
```
Comment (Pugzy) [inline docs/modules/general/introduction.mdx:24]: to -> of

File: docs/modules/general/introduction.mdx (hunk 1 above)
Comment (Pugzy) [inline docs/modules/general/introduction.mdx:8]: started off in 2012 -> started in 2012

File: docs/modules/general/introduction.mdx [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+This plugin could manage the match and reset the map easily at the end of the game.
+Other maps were created later, but these all had to be hardcoded into the plugin.
+This was cumbersome and complicated, and there needed to be a new way to quickly program maps into the plugin using an easy to understand language and syntax.
+This problem would later meet its solution, the PvP Game Manager.
+Players unknown to Java programming could now create their own games and matches using an XML file that PGM takes as an input.
+As of now, there are around 1000+ maps made for PGM!
+
+### How it Works
+
+All maps made for PGM are programmed using an XML file called `map.xml`.
```
Comment (Pugzy) [inline docs/modules/general/introduction.mdx:18]: I wouldn't call it `programmed` maybe `configured`.

File: docs/modules/general/introduction.mdx [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+XML is structured and layered silimar to HTML, making it easily readable.
+Everything from objectives, spawn kits, filters, jump pads, build areas, kill rewards and damage disable is defined in a map's XML file.
```
Comment (Pugzy) [inline docs/modules/general/introduction.mdx:20]: seems like a weird list of modules to pick, maybe just teams, spawns, kits, and objectives as the basics

File: docs/modules/general/introduction.mdx (hunk 4 above)
Comment (Pugzy) [inline docs/modules/general/introduction.mdx:24]: > and information can be placed inside and in-between modules

is a bit of a weird way of putting it

File: docs/modules/general/introduction.mdx [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+Most of the XML can be programmed with little involvement of Minecraft, save for testing and getting coordinates.
+PGM is also backwards compatible with XML files that were made for a previous spec (map proto) so older maps don't need to be upgraded to be playable.
+
+Aspects to a map are defined by modules, and information can be placed inside and in-between modules.
+
+Take a look at an [example](/docs/examples/airship-battle) to see how the XML works and how the modules work together.
+
+The rest of this documentation will explain how to use the modules to build a working XML file for your map.
+Any text editor of your choice can be used to write XML files.
```
Comment (Pugzy) [inline docs/modules/general/introduction.mdx:29]: would possibly recommend specific ones with XML syntax support

**Review by Pugzy — COMMENTED** [review]
A few spelling and grammar issues (thanks Grammarly) and a few bits that don't quite seem 💯 .

File: docs/modules/general/introduction.mdx (hunk 6 above)
Comment (CoWinkKeyDinkInc) [reply docs/modules/general/introduction.mdx]: I think most people would understand the concepts of a kill reward or jump pads, I just wanted to show that even more sophisticated stuff can be easily set.

**Comment by CoWinkKeyDinkInc** [comment]
I forgot to install hunspell on my new Arch install lol

**Comment by Indicardo** [comment]
Hmm, looks great but I think it might too short. Can you think of any other details worth mentioning? I was thinking about a paragraph or two explaining how PGM lets you play almost anything thanks to its versatility, but that's just an idea

**Comment by CoWinkKeyDinkInc** [comment]
I felt it was a bit short but ran out of material at the time, I think I picked up some new stuff after looking at what Electroid wrote for PGM's README.

File: docs/modules/general/introduction.mdx [hunk 8]
```diff
@@ -6,24 +6,34 @@ title: Introduction
...
 PGM originally started in 2012 as a single game plugin specifically built to run one map, Airship Battle.
-This plugin could manage the match and reset the map easily at the end of the game.
+This plugin could manage the match and reset the map easily at the end of the game, which was unique at the time as there were no comprehensive Bukkit plugins that could manage PvP matches.
 Other maps were created later, but these all had to be hard-coded into the plugin.
 This was cumbersome and complicated, and there needed to be a new way to quickly program maps into the plugin using an easy to understand language and syntax.
 This problem would later meet its solution, the PvP Game Manager.
 Players unknown to Java programming could now create their own games and matches using an XML file that PGM takes as an input.
-As of now, there are over 1000 maps made for PGM!
+This was the signature plugin that was exclusive to a major Minecraft server called Overcast Network (also known as oc.tc).
+After the server's clousure, it was open sourced for the community to enjoy.
```
Comment (Indicardo) [inline docs/modules/general/introduction.mdx:15]: closure

File: docs/modules/general/introduction.mdx [hunk 9]
```diff
@@ -6,24 +6,34 @@ title: Introduction
...
+A new server called Stratus Network became the lead maintainer of PGM, adding new features to it, but later on they became unable to maintain the plugin as it became large and messy.
```
Comment (Indicardo) [inline docs/modules/general/introduction.mdx:16]: Don't think this is really necessary as 1.8 PGM is an early fork from the original Overcast PGM code

File: docs/modules/general/introduction.mdx [hunk 10]
```diff
@@ -6,24 +6,34 @@ title: Introduction
...
+This current iteration of PGM has been rebuilt to heavily simplify its scope and has three major changes from the previous versions.
+
+- Based on Minecraft 1.8
+- No backend, website or API
+- Reduced dependencies to make it easier to compile and submit contributions.
+  It is now a lot easier for anyone to run a Minecraft server using PGM and people are still making new maps for it.
```
Comment (Indicardo) [inline docs/modules/general/introduction.mdx:22]: Weird phrasing

File: docs/modules/general/introduction.mdx [hunk 11]
```diff
@@ -6,24 +6,34 @@ title: Introduction
...
 ### How it Works
 
-All maps made for PGM are configured using an XML file called `map.xml`.
+All maps made for PGM are configured using a unique XML file called `map.xml`.
 XML is structured and layered similar to HTML, making it easily readable.
-Everything from objectives, spawn kits, filters, jump pads, build areas, kill rewards is defined in a map's XML file.
+PGM reads `map.xml` when the server is loaded, and when the match starts it applies all the functions that is defined in the XML file onto the map.
+Everything from objectives, spawn kits, filters, jump pads, and kill rewards is defined in a map's XML file.
+Many different gamemodes can be made with PGM, like Team Deathmatch, Capture the Wool, Race, Destroy the Monument, Attack and Defend, and King of the Hill.
+Aspects of a map are defined by modules, and information can be placed inside other modules, making them very versitile and can be used to create many other unique gamemodes and maps.
```
Comment (Indicardo) [inline docs/modules/general/introduction.mdx:32]: versatile

File: docs/modules/general/introduction.mdx [hunk 12]
```diff
@@ -6,24 +6,34 @@ title: Introduction
...
 
 ### How it Works
 
-All maps made for PGM are configured using an XML file called `map.xml`.
+All maps made for PGM are configured using a unique XML file called `map.xml`.
 XML is structured and layered similar to HTML, making it easily readable.
-Everything from objectives, spawn kits, filters, jump pads, build areas, kill rewards is defined in a map's XML file.
+PGM reads `map.xml` when the server is loaded, and when the match starts it applies all the functions that is defined in the XML file onto the map.
+Everything from objectives, spawn kits, filters, jump pads, and kill rewards is defined in a map's XML file.
+Many different gamemodes can be made with PGM, like Team Deathmatch, Capture the Wool, Race, Destroy the Monument, Attack and Defend, and King of the Hill.
```
Comment (Indicardo) [inline docs/modules/general/introduction.mdx:31]: can be reproduced? I don't know, "can be made" is just weird

File: docs/modules/general/introduction.mdx [hunk 13]
```diff
@@ -6,24 +6,34 @@ title: Introduction
...
+A new server called Stratus Network became the lead maintainer of PGM, adding new features to it, but later on they became unable to maintain the plugin as it became large and messy.
+This current iteration of PGM has been rebuilt to heavily simplify its scope and has three major changes from the previous versions.
+
+- Based on Minecraft 1.8
+- No backend, website or API
+- Reduced dependencies to make it easier to compile and submit contributions.
+  It is now a lot easier for anyone to run a Minecraft server using PGM and people are still making new maps for it.
+  As of now, there are over 1000 maps made for PGM!
 
 ### How it Works
```
Comment (Indicardo) [inline docs/modules/general/introduction.mdx:25]: I think this section should include a link to the [Main Map Element](https://pgm.dev/docs/modules/general/main) page, as most of the details are explained there

**Review by Indicardo — CHANGES_REQUESTED** [review]
Definitely looks better now, just need to fix some small details 👍

**Comment by CoWinkKeyDinkInc** [comment]
I meant to ask earlier, but I wanted to add an early screenshot of Airship Battle back from 2012 to the history. If anyone can find a good screenshot that would be nice.

File: docs/modules/general/introduction.mdx (hunk 9 above)
Comment (CoWinkKeyDinkInc) [reply docs/modules/general/introduction.mdx:16]: Do you think this part should be removed?

**Comment by CoWinkKeyDinkInc** [comment]
I found the original YouTube link for Airship Battle, so I linked that instead of a screenshot.

File: docs/modules/general/introduction.mdx (hunk 9 above)
Comment (Indicardo) [reply docs/modules/general/introduction.mdx]: Yea, I would refrain from mentioning other servers (even Overcast, but in this case I think mentioning OCN makes sense)

**Comment by Indicardo** [comment]
Please check https://github.com/PGMDev/Website/pull/25#discussion_r535613828

File: docs/modules/general/introduction.mdx (hunk 9 above)
Comment (CoWinkKeyDinkInc) [reply docs/modules/general/introduction.mdx]: Fair enough.

**Comment by CoWinkKeyDinkInc** [comment]
I removed the Status references, i now just mention that the plugin was open sourced, but was complex to use and maintain.

**Comment by Indicardo** [comment]
Looks great, thanks for contributing!

**Pull request merged**